packages/captp/lib/ts-types.d.ts

```diff
@@ -0,0 +1,50 @@
+/* eslint-disable */
+// eslint-disable-next-line spaced-comment
+
+type ERef<T> = PromiseLike<T> | T;
+type Unpromise<T> = T extends ERef<infer U> ? U : T;
+
+export type Syncable<T> = T extends (...args: infer P) => infer R
+  ? (...args: P) => Unpromise<R>
+  : T extends Record<string | number | symbol, Function>
```

Comment:
Aside: I think in general we need to pare this down to `Record<string, Function>` to avoid painting ourselves into a corner. I think we’ll want to handle symbols over CapTP case-by-case and with a more language-neutral treatment.

Reply:
I think there's no corner to paint ourselves into.  This is not an expression of what can be marshalled more than it is a type constraint that TS already has.  Records are by default `Record<string | number | symbol, any>`.  The definition of `Sync` should not necessarily reflect the constraints of the specific serialisation system.